flink-yarn/src/main/java/org/apache/flink/yarn/Utils.java

```diff
@@ -149,61 +155,79 @@ public static void setupYarnClassPath(Configuration conf, Map<String, String> ap
 		String appId,
 		Path localSrcPath,
 		Path homedir,
-		String relativeTargetPath) throws IOException {
-
-		File localFile = new File(localSrcPath.toUri().getPath());
-		if (localFile.isDirectory()) {
-			throw new IllegalArgumentException("File to copy must not be a directory: " +
-				localSrcPath);
-		}
+		String relativeTargetPath,
+		@Nullable FileStatus preUploadedFileStatus) throws IOException {
```

Comment:
I think we should separate register pre-uploaded files into another method instead just hacking a huge if-else block. If you take a close look you will find this two code path share little code and pre-uploaded path even doesn't use most of the parameter.

Reply:
I have moved the `setupLocalResource` without preUploadedFileStatus to another method to  avoid huge if-else block.

Reply:
It seems that you don't push the commit so far. One reminder is for smooth review you can push the change in another commit :-)

Reply:
+1, how about splitting it into: `setUpLocalResource()` and `setUpPreloadedResource` --> since `Local` really referring to something that's "local" to the submission client that runs this `Utils.class`

BTW I really like how @wangyang0918 split the setup and the register part out! e.g. `registerLocalResource` (renaming it to `registerResource` might've been even better)

Reply:
Thanks @TisonKun and @walterddr. It makes sense to me. I will split in into two methods `setupLocalResource()` and `setupPreUploadedResource`.

However, i suggest not to change the name `registerLocalResource`. Since the local not means the local file, it means the Yarn `LocalResource`. It is a proper noun of Yarn.

```
LocalResource localResource = Records.newRecord(LocalResource.class);
```